packages/store/src/internal/state-operations.ts

```diff
@@ -12,39 +11,39 @@ import { deepFreeze } from '../utils/freeze';
  */
 @Injectable()
 export class InternalStateOperations {
-  constructor(
-    private _stateStream: StateStream,
-    private _dispatcher: InternalDispatcher,
-    private _config: NgxsConfig
-  ) {}
+  private stateStream: StateStream;
+  private dispatcher: InternalDispatcher;
+  private readonly developmentMode: boolean;
 
-  /**
-   * Returns the root state operators.
-   */
-  getRootStateOperations(): StateOperations<any> {
-    const rootStateOperations = {
-      getState: () => this._stateStream.getValue(),
-      setState: newState => this._stateStream.next(newState),
-      dispatch: actions => this._dispatcher.dispatch(actions)
-    };
+  constructor(context: Injector) {
+    this.stateStream = context.get(StateStream);
+    this.dispatcher = context.get(InternalDispatcher);
+    this.developmentMode = context.get(NgxsConfig).developmentMode;
+    InternalStateOperations.checkDevelopmentMode(this.developmentMode);
+  }
 
-    if (this._config.developmentMode) {
-      return this.ensureStateAndActionsAreImmutable(rootStateOperations);
```

Comment:
The compositional pattern used here was deliberate please don't move away from it.

Reply:
@markwhitfeld but the code is cleaner
![image](https://user-images.githubusercontent.com/12021443/47557355-243a2400-d919-11e8-992a-6dd2480ccf21.png)